src/infra/ingest-datafusion/src/visitor.rs

```diff
@@ -52,35 +53,32 @@ pub struct DataWriterDataFusionMetaDataStateVisitor<'a> {
 }
 
 impl<'a> DataWriterDataFusionMetaDataStateVisitor<'a> {
-    pub fn new(head: Multihash, maybe_source_name: Option<&'a str>) -> (Decision, Self) {
-        let next_block_flags = Flag::SET_DATA_SCHEMA
-            | Flag::ADD_DATA
-            | Flag::SET_POLLING_SOURCE
-            | Flag::DISABLE_POLLING_SOURCE
-            | Flag::ADD_PUSH_SOURCE
-            | Flag::DISABLE_PUSH_SOURCE
-            | Flag::SET_VOCAB
-            | Flag::SEED;
-
-        (
-            Decision::NextOfType(next_block_flags),
-            Self {
-                head,
-                maybe_source_name,
-
-                next_block_flags,
-                data_slices: Vec::new(),
-
-                maybe_schema: None,
-                maybe_source_event: None,
-                maybe_set_vocab: None,
-
-                maybe_prev_checkpoint: None,
-                maybe_prev_watermark: None,
-                maybe_prev_source_state: None,
-                maybe_prev_offset: None,
-            },
-        )
+    pub fn new(head: Multihash, maybe_source_name: Option<&'a str>) -> Self {
```

Comment:
This visitor ended up very complex because there are at lest 5 different things that writer needs:
- last `AddData` event to connect next `AddData` event to
- names of parquet files from all `AddData` (for processing merge strategies)
- polling or push source
- data schema
- vocab

Therefore instead of having one very complex visitor you could express this as accepting 5 simple visitors that focus on one job each.

Reply:
Explicitly pieces were extracted in individual Visitors -- Done